File: src/main/scala/mffs/item/module/projector/ItemModuleDome.java
package mffs.item.module.projector;

import java.util.Iterator;
import java.util.Set;

import mffs.item.module.ItemModule;
import net.minecraft.tileentity.TileEntity;
import universalelectricity.api.vector.Vector3;
import calclavia.api.mffs.IFieldInteraction;

public class ItemModuleDome extends ItemModule
{
	public ItemModuleDome(int i)
	{
		super(i, "moduleDome");
		this.setMaxStackSize(1);
	}

	@Override
	public void onCalculate(IFieldInteraction projector, Set<Vector3> fieldBlocks)
	{
		Vector3 absoluteTranslation = new Vector3((TileEntity) projector).translate(projector.getTranslation());

		Iterator<Vector3> it = fieldBlocks.iterator();

		while (it.hasNext())
		{
			Vector3 position = it.next();

			if (position.y < absoluteTranslation.y)
			{
				it.remove();
			}
		}
	}
}

File: src/main/scala/mffs/item/module/projector/ItemModuleDome.scala
package mffs.item.module.projector

import java.util.Set
import mffs.item.module.ItemModule
import net.minecraft.tileentity.TileEntity
import universalelectricity.core.transform.vector.Vector3
import resonant.api.mffs.IFieldInteraction
import scala.collection.convert.wrapAll._

class ItemModuleDome(id: Int) extends ItemModule(id, "moduleDome")
{
	setMaxStackSize(1)

	override def onCalculate(projector: IFieldInteraction, fieldBlocks: Set[Vector3])
	{
		val absoluteTranslation = new Vector3(projector.asInstanceOf[TileEntity]).translate(projector.getTranslation)
		val newField = fieldBlocks.par.filter(_.y > absoluteTranslation.y).seq
		fieldBlocks.clear()
		fieldBlocks.addAll(newField)
	}
}